Sources/MapboxNavigation/WayNameView.swift

```diff
@@ -10,10 +10,43 @@ open class WayNameLabel: StylableLabel {}
 /// :nodoc:
 @objc(MBWayNameView)
 open class WayNameView: UIView {
-    private static let textInsets = UIEdgeInsets(top: 6, left: 14, bottom: 6, right: 14)
+    private static let textInsets = UIEdgeInsets(top: 3, left: 14, bottom: 3, right: 14)
     
     lazy var label: WayNameLabel = .forAutoLayout()
     
+    /**
+     Allows to completely hide `WayNameView`.
+     */
+    public var isAlwaysHidden: Bool = false {
+        didSet {
+            isHidden = isAlwaysHidden
+            setNeedsDisplay()
+        }
+    }
+    
+    var _isHidden: Bool = false
+    
+    /**
+     Allows to hide or show `WayNameView`. In case if `WayNameView.isAlwaysHidden` is set to `true`
+     modification of this value will have no effect and `WayNameView` will remain hidden.
+     */
+    open override var isHidden: Bool {
```

Comment:
The separate `isHidden` and `isAlwaysHidden` properties are a bit unintuitive. `isHidden` is normally a way for a developer to hide the view without it automatically coming back. We can accomplish that by embedding another view within WayNameView that we automatically hide when there’s no road name to display. Alternatively, we should introduce a property `visibility` of type `OrnamentVisibility` (from the map SDK) that allows for an `adaptive` visibility by default.

/ref https://github.com/mapbox/mapbox-navigation-ios/pull/3534#discussion_r739372134

Reply:
`isAlwaysHidden` was introduced to have similarity with public property of `SpeedLimitView.isAlwaysHidden`. Not sure that usage of `OrnamentVisibility` will help us here, we'll basically use only `.adaptive` and `.hidden` cases, `.visible` will not be relevant. 

Wrapping `WayNameView` in another view, which can be ultimately hidden leaves us with breaking API changes as we currently expose `WayNameView` on iOS as well, and it'd make sense to be able to hide it as well there. I'll do some more tests though to see if we can use such approach.

Reply:
In the end I've decided to go with the wrapping approach. Please check the latest changes. Hopefully I did not bring any regression, but now to completely hide `WayNameView` we can use `isHidden` property.